components/higher-order/with-dragging/index.js

```diff
@@ -0,0 +1,163 @@
+			const element = document.getElementById( elementId );
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			if ( element && cloneWrapper ) {
+				// Remove clone.
+				element.parentNode.removeChild( cloneWrapper );
+			}
+
+			// Reset cursor.
+			document.body.classList.remove( 'dragging' );
+
+			this.setState( { elementId: null, cloneNodeId: null } );
+
+			document.removeEventListener( 'dragover', this.onDragOver );
+			event.stopPropagation();
+		}
+
+		/*
+		 * Updates positioning of element clone based on mouse movement during dragging.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 */
+		onDragOver( event ) {
+			const { cloneNodeId } = this.state;
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			cloneWrapper.style.top =
+				`${ parseInt( cloneWrapper.style.top, 10 ) + parseInt( event.clientY, 10 ) - parseInt( this.cursorTop, 10 ) }px`;
+			cloneWrapper.style.left =
+				`${ parseInt( cloneWrapper.style.left, 10 ) + parseInt( event.clientX, 10 ) - parseInt( this.cursorLeft, 10 ) }px`;
+
+			// Update cursor coordinates.
+			this.cursorLeft = event.clientX;
+			this.cursorTop = event.clientY;
+		}
+
+		/**
+		 *  - Clones the current element and spawns clone over original element.
+		 *  - Adds a fake temporary drag image to avoid browser defaults.
+		 *  - Sets transfer data.
+		 *  - Adds dragover listener.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 * @param  {string} uid       The unique identifier of the element to be dragged.
+		 * @param  {string} elementId The HTML id of the element to be dragged.
+		 * @param  {number} order     The current index of the element to be dragged.
+		 * @param  {string} type      A unique constant identifying the type of dragging operation.
+		 */
+		onDragStart( event, uid, elementId, order, type ) {
+			const element = document.getElementById( elementId );
+			const elementWrapper = element.parentNode;
+			const elementRect = element.getBoundingClientRect();
+			const elementTopOffset = parseInt( elementRect.top, 10 );
+			const elementLeftOffset = parseInt( elementRect.left, 10 );
+			const clone = element.cloneNode( true );
+			const cloneWrapper = document.createElement( 'div' );
+			const cloneNodeId = `clone-wrapper-${ element.id }`;
+			const dragImage = document.createElement( 'div' );
+
+			this.setState( { cloneNodeId, elementId } );
+
+			// Set a fake drag image to avoid browser defaults. Remove from DOM right after.
+			if ( 'function' === typeof event.dataTransfer.setDragImage ) {
+				dragImage.id = `drag-image-${ element.id }`;
+				dragImage.classList.add( this.config.dragImageClass );
+				document.body.appendChild( dragImage );
+				event.dataTransfer.setDragImage( dragImage, 0, 0 );
+				setTimeout( ( ( _dragImage ) => () => {
+					document.body.removeChild( _dragImage );
+				} )( dragImage ), 0 );
+			}
+
+			event.dataTransfer.setData(
+				'text',
+				JSON.stringify( {
+					uid: uid,
```

Comment:
Minor: ES6 object property shorthand allows you to shorten this to:

```js
JSON.stringify( {
	obj,
	// ...
} );
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015